rpc/src/rpc.rs

```diff
@@ -450,10 +493,18 @@ impl JsonRpcRequestProcessor {
         })?;
         let encoding = encoding.unwrap_or(UiAccountEncoding::Base64);
 
-        let accounts = pubkeys
-            .into_iter()
-            .map(|pubkey| get_encoded_account(&bank, &pubkey, encoding, data_slice))
-            .collect::<Result<Vec<_>>>()?;
+        let mut accounts = Vec::with_capacity(pubkeys.len());
+        for pubkey in pubkeys {
+            let bank = Arc::clone(&bank);
+            accounts.push(
+                self.runtime
+                    .spawn_blocking(move || {
+                        get_encoded_account(&bank, &pubkey, encoding, data_slice)
+                    })
+                    .await
+                    .expect("rpc: get_encoded_account panicked")?,
```

Comment:
I think we should handle any errors nicely, shouldn't we?

Reply:
My thinking was: spawn_blocking returns Err if the spawned code panics. If the spawned code panics, it sounds like it's a bug? Not sure it makes sense to handle gracefully?

Reply:
it also returns error upon cancellation

Reply:
I don't think it can in this case? We're creating the JoinHandle and immediately awaiting it. We're not calling abort() and we're not giving out any AbortHandle(s), so nobody can cancel?

Reply:
the runtime won't cancel outstanding tasks when its threads catch SIGINT?

Reply:
I don't think so? https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#shutdown

> Blocking functions spawned through [Runtime::spawn_blocking](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.spawn_blocking) keep running until they return.

Tokio cancels tasks by stopping to poll them, so they stop at the next await point. But things inside spawn_blocking don't await, so they can only run until completion (or process exit). pthread_cancel isn't a thing etc